docs/internals-object-definition.md

```diff
@@ -0,0 +1,123 @@
+# Internals: Types of Object Definitions
+
+An Object Definition defines a Type of Thing - for example a String, a List of a String, a List of a List of a List of an Object. Different types of Object Definition exist within Pandora, these are:
+
+1. Options Object Definition - defines values that can be sent as part of an HTTP Request (Headers/QueryString).
+2. ObjectDefinition - defines the type of a Field within a Model.
+3. Terraform Schema Field Type - to define the type of a Field in Terraform.
+
+## Options Object Definition
+
+HTTP Operations can support Options, which are values sent at Request time, either in the HTTP Headers and/or the QueryString.
+
+At this point in time these support the following Basic types:
+
+* Booleans
+* CSVs
+* Float (the Swagger defines Double/Decimal/Number - which we normalize into a Float)
+* Integers
+* List
+* Strings
+
+Examples:
+
+* A Boolean
+* A CSV of a String
+* A List of a String
+* A String
+
+Limitations:
+
+* A List cannot contain a List
+
+Notes:
+
+* At present the Object Definition type is reused, but shouldn't be, as we'll want to extend this in the future when we parse Data Plane (which has way more possible values for types).
```

Comment:
Could we be explicit about what specifically is missing here? e.g.
* add an Options Object Definition type
* add a Terraform Object Definition type